test/indexer/test_indexer_controller.py

```diff
@@ -0,0 +1,139 @@
+        self._test_contribute_and_aggregate()
+
+    def _test_contribution_action_invalid(self):
+        event = [self.generate_record(action='banana', notification='slug', catalog=self.catalog)]
+        self.assertRaises(AssertionError, self.controller.contribute, event)
+
+    def _test_contribution_action_reindex(self):
+        event = [self.generate_record(action='reindex',
+                                      prefix='ff')]
+        with mock.patch('azul.azulclient.AzulClient.do_remote_reindex') as acr:
+            acr.return_value = True
+            self.controller.contribute(event)
+            self.assertTrue(acr.called)
+
+    @mock_sts
+    @mock_sqs
+    def _test_contribute_and_aggregate(self):
+        """
+        Index two bundles that make contributions to the same project.
+        Inspect that the contributions match the tallies that are returned to
+        SQS. During aggregation only the project entity is deferred due to
+        multiple contributions.
+        """
+        self._create_mock_queues()
+        event = list()
+        bundles = list()
+        list_of_contributions = list()
+        bundle_fqids = [BundleFQID('56a338fe-7554-4b5d-96a2-7df127a7640b', '2018-03-28T151023.074974Z'),
+                        BundleFQID('b2216048-7eaa-45f4-8077-5a3fb4204953', '2018-03-29T104041.822717Z')]
+        for bundle_fqid in bundle_fqids:
+            event.append(self.generate_record(
+                action='add',
+                catalog=self.catalog,
+                notification=self.client.synthesize_notification(self.catalog, bundle_fqid)))
+            bundle = self._load_canned_bundle(bundle_fqid)
+            bundles.append(bundle)
+            list_of_contributions.append(self.index_service.transform(self.catalog,
+                                                                      bundle=bundle,
+                                                                      delete=False))
+        with mock.patch('azul.indexer.index_controller.IndexController.transform') as mock_transform, \
+            mock.patch('azul.indexer.index_controller.IndexController.index_service',
+                       new_callable=mock.PropertyMock) as mock_index_service:
+            mock_index_service.return_value = ForcedRefreshIndexService()
+            mock_transform.side_effect = list_of_contributions
+            self.controller.contribute(event)
+            tallies = list(json.loads(x.body) for x in
+                           queues.Queues(delete=True).read_messages(self.controller._tallies_queue()))
+            entities_from_tallies = set((x["entity_id"], x["entity_type"]) for x in tallies)
+            entities_from_contributions = set((contributions.entity.entity_id, contributions.entity.entity_type)
+                                              for cl in list_of_contributions for contributions in cl)
+            self.assertSetEqual(entities_from_tallies, entities_from_contributions)
+            event.clear()
```

Comment:
```suggestion
            event = []
```

Reply:
Is there a reason to prefer one over the other in this context?